tracing/src/main/java/com/palantir/tracing/Trace.java

```diff
@@ -30,24 +30,35 @@
  */
 public final class Trace {
 
+    private static final Trace NOOP = new Trace(false, "noop");
+
     private final Deque<OpenSpan> stack;
     private final boolean isObservable;
     private final String traceId;
 
-    private Trace(ArrayDeque<OpenSpan> stack, boolean isObservable, String traceId) {
+    private Trace(Deque<OpenSpan> stack, boolean isObservable, String traceId) {
         checkArgument(!traceId.isEmpty(), "traceId must be non-empty");
 
         this.stack = stack;
         this.isObservable = isObservable;
         this.traceId = traceId;
     }
 
+    static Trace create(boolean isObservable, CharSequence traceId) {
+        if (isObservable) {
+            return new Trace(isObservable, traceId.toString());
+        }
+        return NOOP;
+    }
+
     Trace(boolean isObservable, String traceId) {
```

Comment:
Wanna mark this @VisibleForTesting and make `DeferredTracer` call your new static `create` method?